apps/src/lib/client/rpc.rs

```diff
@@ -119,12 +133,44 @@ pub async fn query_balance(ctx: Context, args: args::QueryBalance) {
             let owner = ctx.get(&owner);
             let mut found_any = false;
             for (token, currency_code) in tokens {
-                let key = token::balance_key(&token, &owner);
-                if let Some(balance) =
-                    query_storage_value::<token::Amount>(&client, &key).await
-                {
-                    println!("{}: {}", currency_code, balance);
-                    found_any = true;
+                let prefix = token.to_db_key().into();
+                let balances = query_storage_prefix::<token::Amount>(
+                    client.clone(),
+                    prefix,
+                )
+                .await;
+                if let Some(balances) = balances {
+                    let stdout = io::stdout();
+                    let mut w = stdout.lock();
+                    for (key, balance) in balances {
+                        match token::is_any_multitoken_balance_key(&key) {
+                            Some((sub_prefix, o)) if *o == owner => {
+                                writeln!(
+                                    w,
+                                    "{} with {}: {}",
+                                    currency_code, sub_prefix, balance
+                                )
+                                .unwrap();
+                                found_any = true;
+                            }
+                            Some(_) => {}
+                            None => {
+                                if let Some(o) =
+                                    token::is_any_token_balance_key(&key)
+                                {
+                                    if *o == owner {
+                                        writeln!(
+                                            w,
+                                            "{}: {}",
+                                            currency_code, balance
+                                        )
+                                        .unwrap();
+                                        found_any = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
```

Comment:
This code block exists three times identically.  Maybe use a helper func

Reply:
I added `print_balances` func.